### DISCUSSION TITLE
How to vertically mark text / add to selection?

### DISCUSSION THREAD

**Discussion by Robert-M-Muench** [helix-editor/helix#949] [Q&A] [answered]
What's the quickest way when I need to insert the same text at the current cursor position and the next X lines down/up at the same column? I'm missing a vertical mark command.

I currently use `/` and `N` to search and add to the selection. But this is tedious because the hit where the cursor currently is located is not considered, I need to find a good query to avoid highlighting other text parts, etc. 

Simply marking and moving the cursor up/down would be better.

**Comment by sudormrfbin** [comment, marked as answer]
You can use `C` and `Alt-C` to add cursors below and above the primary cursor.

**Reply by Robert-M-Muench** [reply]
Thanks. That wasn't clear from the description on the keymap page:

* C | Copy selection onto the next line | copy_selection_on_next_line
* Alt-C | Copy selection onto the previous line | copy_selection_on_prev_line

I understand *Copy Selection* as copying the selected text to the next line. I think changing it to *Add cursor...* as you wrote would help to make that more clear.

**Reply by sudormrfbin** [reply]
`C` actually replicates the selection of the same width on the next line, so it probably says "copy" because of that. I do agree that it doesn't seem very intuitive at first glance.

**Reply by pickfire** [reply]
Do you have any suggestion how can we name it better so it is easier for others to understand?

**Reply by Robert-M-Muench** [reply]
I think, *Replicate current selection onto the next/previous line* hits it pretty clear. Or maybe, *Repeat current selection form/structure/shape onto next/previous line*.

**Reply by pickfire** [reply]
What about *repeat selection to next line*?

**Reply by sudormrfbin** [reply]
I changed it in https://github.com/helix-editor/helix/pull/969 a while back and thought it will be linked back from here like in issues (simply says "Add cursor above/below" in addition to the previous description).
